core/src/test/scala/eu/shiftforward/apso/akka/http/ProxySupportSpec.scala

```diff
@@ -153,6 +156,14 @@ class ProxySupportSpec(implicit ee: ExecutionEnv) extends Specification with Spe
         status == OK
         responseAs[String] must be_==("/remote-proxy")
       }
+
+      proxy.sendRequest(Get("/proxied"), failOnDrop = false).flatMap {
+        case Complete(res) => res.entity.toStrict(10.seconds).map { r =>
+          val charset = r.contentType.charsetOption.map(_.nioCharset()).getOrElse(Charset.forName("UTF-8"))
```

Comment:
I wonder if for testing purposes you can't just call `r.data.utf8String` instead of doing the full, "correct" process.